aten/src/ATen/native/RNN.h

```diff
@@ -19,5 +19,36 @@ DECLARE_DISPATCH(rnn_packed_fn, gru_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_tanh_packed_cudnn_stub);
 DECLARE_DISPATCH(rnn_packed_fn, rnn_relu_packed_cudnn_stub);
 
-}} // namespace at::native
 
+void check_device(const Tensor& input, const TensorList& params, const TensorList& hiddens) {
```

Comment:
Why do we have additional helper here, and not use checkAllSameGPU https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorUtils.cpp#L131 ?
Also, agree with Adam, checks should be only in cudnn RNN, everything else in that native paths is checked already.

Reply:
Thanks for this great suggestion! Totally agree that `checkAllSameGPU` serve the purpose already! I will use it instead.